### PR TITLE
Add support for service registration and deregistration.

### PR DESCRIPTION
**Pull request by michaeldejong**
Allows us to register and deregister services to and from Consul.

Currently we do assign a random UUID to the `instance` variable in the `ServiceIdentifier` class, if no instance is specified through the environment variables. In that case a service (with a random UUID) is registered to Consul, and failing to deregister that service, will cause it to linger forever in Consul (until someone removes it with an API call).

To combat this, the `INSTANCE` variable should be defined explicitly in the environment variables.

Unfortunately also no tests, since this functionality is directly interacting with Consul and we typically mock these interactions to test higher-level behaviour.

### CODE REVIEW

File: src/main/java/me/magnet/consultant/Consultant.java [hunk 1]
```diff
@@ -316,8 +327,91 @@ private void init() {
...
+		}
+	}
+
+	public void registerService(int port) {
+		String url = host + "/v1/agent/service/register";
+
+		try {
+			String serviceId = null;
+			String serviceName = id.getServiceName();
+			Check check = new Check("http://" + LOCALHOST + ":" + port + "/_health", 10);
```
Comment (AlexKolpa) [inline src/main/java/me/magnet/consultant/Consultant.java]: Could you make that `10` a static field?

**Comment by AlexKolpa** [comment]
Small remark, rest looks good :+1:

File: src/main/java/me/magnet/consultant/Consultant.java [hunk 2]
```diff
@@ -36,6 +45,8 @@
  */
 public class Consultant {
 
+	public static final String LOCALHOST = "127.0.0.1";
```
Comment (TiddoLangerak) [inline src/main/java/me/magnet/consultant/Consultant.java]: This isn't ipv6 compatible. Is that a problem?

File: src/main/java/me/magnet/consultant/Consultant.java [hunk 3]
```diff
@@ -316,8 +327,91 @@ private void init() {
...
+			String serialized = mapper.writeValueAsString(registration);
+
+			HttpPut request = new HttpPut(url);
+			request.setEntity(new StringEntity(serialized));
+			request.setHeader("User-Agent", "Consultant");
+			try (CloseableHttpResponse response = http.execute(request)) {
+				int statusCode = response.getStatusLine().getStatusCode();
+				if (statusCode >= 200 && statusCode < 300) {
+					return;
+				}
```
Comment (TiddoLangerak) [inline src/main/java/me/magnet/consultant/Consultant.java]: `3xx` status codes are just redirects, but there's a good chance that the http client doesn't automatically resolve those. Wouldn't we want to do that anyway?

**Comment by TiddoLangerak** [comment]
:+1:

**Pull request merged**